consensus/tendermint/core/handler.go

```
@@ -220,6 +229,47 @@ eventLoop:
 	c.stopped <- struct{}{}
 }
 
+func (c *core) syncLoop(ctx context.Context) {
+	/*
+		this method is responsible for asking the network to send us the current consensus state
+		and to process sync queries events.
+	*/
+	timer := time.NewTimer(10 * time.Second)
+	c.currentRoundStateMu.RLock()
+	round := c.currentRoundState.Round()
+	height := c.currentRoundState.Height()
+	c.currentRoundStateMu.RUnlock()
+
+	// Ask for sync when the engine starts
+	c.backend.AskSync(c.valSet.Copy())
+
+	for {
+		select {
+		case <-timer.C:
+			c.currentRoundStateMu.RLock()
+			currentRound := c.currentRoundState.Round()
+			currentHeight := c.currentRoundState.Height()
+			c.currentRoundStateMu.RUnlock()
+			// we only ask for sync if the current view stayed the same for the past 10 seconds
+			if currentHeight.Cmp(height) == 0 && currentRound.Cmp(round) == 0 {
+				c.backend.AskSync(c.valSet.Copy())
+			}
+			round = currentRound
+			height = currentHeight
+			timer = time.NewTimer(10 * time.Second)
+		case ev, ok := <-c.syncEventSub.Chan():
+			if !ok {
+				return
+			}
+			event := ev.Data.(events.SyncEvent)
+			log.Info("Processing sync message", "from", event.Addr)
+			c.SyncPeer(event.Addr)
+		case <-ctx.Done():
+			return
```

Why do we ignore the error?

---

Only thing I see I can do is to print it?

---

Maybe we could check if ctx.Err() == context.Cancelled , and if not than log an error